runtime/builtin_types.go

```diff
@@ -617,6 +617,85 @@ func builtinSorted(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
 	return result, nil
 }
 
+func builtinSum(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+	argc := len(args)
+	expectedTypes := []*Type{ObjectType, ObjectType}
+	if argc == 1 {
+		expectedTypes = expectedTypes[:1]
+	}
+	if raised := checkFunctionArgs(f, "sum", args, expectedTypes...); raised != nil {
+		return nil, raised
+	}
+	iter, raised := Iter(f, args[0])
+	if raised != nil {
+		return nil, raised
+	}
+	var result *Object
+	if argc > 1 {
+		if args[1].typ == StrType {
+			return nil, f.RaiseType(TypeErrorType, "sum() can't sum strings [use ''.join(seq) instead]")
+		}
+		result = args[1]
+	} else {
+		result = NewInt(0).ToObject()
+	}
+	if result.typ == IntType {
+		item, raised := Next(f, iter)
+		for ; raised == nil; item, raised = Next(f, iter) {
+			if item.typ == IntType {
+				result = NewInt(toIntUnsafe(result).Value() + toIntUnsafe(item).Value()).ToObject()
+				continue
+			}
+			result, raised = Add(f, result, item)
+			break
+		}
+		if raised != nil {
+			if raised.isInstance(StopIterationType) {
+				f.RestoreExc(nil, nil)
+				return result, nil
+			}
+			return nil, raised
+		}
+	}
+	if result.typ == FloatType {
```

Comment:
This logic is from CPython, and i think it's trying to not create new python object in the middle while all the elements is int or float, and i think grumpy is having the same problem if we do not treat them special.

But i don't know should we treat longt type special like these, can you give me a suggestion?